db_checker.py

```diff
@@ -114,24 +116,20 @@ def allowed_unit(self, unit):
         """
         This method checks that the given unit conforms to standard
         """
-        if unit in allowed_units:
-            return True
+        # expand macro $(A) to a valid unit, expand $(A=B) to B
+        u = re.sub(r'\$[({].*?=(.*)?[})]', r'\1', unit)
+        u = re.sub(r'\$[({].*?[})]', 'm', u)
 
-        # otherwise check for macro
-        if "$" in unit:
-            return True
+        # split unit amalgamations
+        units = re.split(r'[/ ()]', u)
 
-        # otherwise split unit amalgamations
-        units = re.split(r'[/ ()]', unit)
         for u in units:
+            # remove any powers of units e.g. mm^2 -> mm
+            u = re.sub(r'\^[-]?\d', '', u)
```

Comment:
The changes cover most of these cases now, but still pass if there is just one power modifier somewhere in the middle of the unit (as long as there's at least one letter that comes before it). Adding an "end of string" character `$` to the end of the regex should do the trick though (assuming that the power modifier will always be the last part of the unit description)